src/commands/moderators/deleteAllMessage.ts

```diff
@@ -76,7 +76,12 @@ export default defineCommandHandler({
     let numDeleted = 0
     const { client } = botContext
     for (const [channelId, channel] of client.channels.cache) {
-      if (channel.type === ChannelType.GuildText) {
+      if (
+        channel.type === ChannelType.GuildText || //for text channel
+        channel.type === ChannelType.GuildVoice || //for openchat at voice channel
+        channel.type === ChannelType.GuildStageVoice || //for openchat stage channel
+        channel.type === ChannelType.PublicThread //for public thread and forum
+      ) {
```

Comment:
Can we extract it to variable named like `supportedTextChannel`

```suggestion
      const supportedTextChannel: ChannelType[] = [ChannelType.GuildText, ...] // move this to top level of the file is also suggested
      if (supportedTextChannel.includes(channel.type)) {
```

Reply:
it can't guarantee that 'channel' will have the 'messages'or other property and leading to the type error

Reply:
it can't guarantee that 'channel' will have the 'messages'or other property so it will leading to the type error
btw, is any idea to resolve this
![ShareX_n1zrCOgjXZ_65](https://github.com/creatorsgarten/kaogeek-discord-bot/assets/67526393/ed9a339e-e4dc-4f28-849f-e8d054e10c58)

Reply:
Hmm, could you do something like [this reply](https://github.com/microsoft/TypeScript/issues/31018#issuecomment-808135421)? Just copy the function and put it in utility folder is fine. I think we will encounter again this problem in the future.